### PR TITLE
Adding Benchmarks back into agbench

### PR DESCRIPTION
**Pull request by husseinmozannar**
## Why are these changes needed?

Added benchmarks back into autogen, notably GAIA, AssistantBench, HumanEval and WebArena. Renamed everything to be consistent with new package, enabled creation of endpoint from environment file.

Modified readme of magentic-one with a few changes regarding installation instructions

Fixed renaming issues in agbench to be consistent with new package structure.

## Checks

- [ x] ran AssistantBench, GAIA and HumanEval on a few examples, but not entirely. So far behavior as expected. 
- [] no testing for WebArena so far

## TO-DO

- [x] formatting checks and others
- [x] double check from magentic-one team

### CODE REVIEW

**Comment by husseinmozannar** [comment]
Might have renamed things outside of my reach by mistake, can revert those (but most seem to make sense)

File: python/packages/autogen-magentic-one/readme.md
Comment (gagb) [inline python/packages/autogen-magentic-one/readme.md:1]: can you summarize what changed in this file? If not, remove from commit.

**Review by gagb — APPROVED** [review]

**Comment by husseinmozannar** [comment]
Will continue working on agbench for a while. The documentation is outdated, and there are some important features that are necessary which I am still implementing.
